src/app/_components/ListboxButton.tsx

```diff
@@ -14,13 +14,13 @@ export function ListboxButton({
   children,
 }: ListboxButtonProps) {
   return (
-    <Listbox.Button
+    <HeadlessUIListboxButton
       aria-haspopup="listbox"
       aria-expanded={open}
       aria-label={ariaLabel}
-      className="focus:brand-outline inline-flex w-full items-center justify-between gap-2 rounded-lg border border-brand-300 p-3 text-brand-300 hover:border-current hover:text-brand-400 md:min-w-40"
+      className="inline-flex w-full items-center justify-between gap-2 rounded-lg border border-brand-300 p-3 text-brand-300 focus:brand-outline hover:border-current hover:text-brand-400 md:min-w-40"
     >
```

Comment:
I'm not sure we still need all these props.

Looking at the type definition `ListboxButtonProps`, they seem to manage the aria attributes for us:
```ts
type ButtonPropsWeControl = 'aria-controls' | 'aria-expanded' | 'aria-haspopup' | 'aria-labelledby' | 'disabled';
```

Also, `ListboxButtonProps` could extend the type definition exported by `'@headlessui/react'`

https://headlessui.com/react/button